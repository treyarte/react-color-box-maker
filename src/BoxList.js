import React, { useState } from 'react';
import { v4 as uuid } from 'uuid';
import './BoxList.css';
import Box from './Box';
import NewBoxForm from './NewBoxForm';

const BoxList = () => {
  const INITIAL_STATE = [];
  const [boxes, setBoxes] = useState(INITIAL_STATE);

  const addBox = (box) => {
    const newBox = { ...box, id: uuid() };
    setBoxes((box) => [...boxes, newBox]);
  };

  const remove = (id) => {
    setBoxes(boxes.filter((box) => box.id !== id));
  };
  return (
    <div className='box-list'>
      <div>
        <h3>BoxList</h3>
      </div>
      <div className='box-form'>
        <NewBoxForm addBox={addBox} />
      </div>
      <div className='boxes'>
        {boxes.map(({ id, width, height, color }) => (
          <Box
            id={id}
            color={color}
            height={height}
            width={width}
            key={id}
            remove={remove}
          />
        ))}
      </div>
    </div>
  );
};

export default BoxList;
