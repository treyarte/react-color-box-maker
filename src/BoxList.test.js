import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import BoxList from './Boxlist';

it('Should render without crashing', () => {
  render(<BoxList />);
});

it('Should match its snapshot', () => {
  const { asFragment } = render(<BoxList />);

  expect(asFragment).toMatchSnapshot();
});

it('Should create a new box', () => {
  const { getByLabelText, queryByText, queryByDisplayValue } = render(
    <BoxList />
  );

  const colorInput = getByLabelText('Color');
  const widthInput = getByLabelText('Width');
  const heightInput = getByLabelText('Height');
  const submitBtn = queryByText('Add Box');

  fireEvent.change(colorInput, { target: { value: 'pink' } });
  fireEvent.change(widthInput, { target: { value: 300 } });
  fireEvent.change(heightInput, { target: { value: 500 } });
  fireEvent.click(submitBtn);

  expect(queryByText('Remove')).toBeInTheDocument();
  expect(queryByText('Remove').previousSibling).toHaveStyle(`
    width: 300px;
    height: 500px;
    background-color: pink;
  `);
});

it('Should remove a box', () => {
  const { getByLabelText, queryByText, queryByDisplayValue } = render(
    <BoxList />
  );

  const colorInput = getByLabelText('Color');
  const widthInput = getByLabelText('Width');
  const heightInput = getByLabelText('Height');
  const submitBtn = queryByText('Add Box');

  fireEvent.change(colorInput, { target: { value: 'pink' } });
  fireEvent.change(widthInput, { target: { value: 300 } });
  fireEvent.change(heightInput, { target: { value: 500 } });
  fireEvent.click(submitBtn);
  const removeBtn = queryByText('Remove');
  expect(removeBtn).toBeInTheDocument();

  fireEvent.click(removeBtn);
  expect(removeBtn).not.toBeInTheDocument();
});
