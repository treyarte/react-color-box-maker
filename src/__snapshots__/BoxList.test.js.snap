// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match its snapshot 1`] = `[Function]`;
